common/protob/messages-zcash.proto

```diff
@@ -0,0 +1,17 @@
+syntax = "proto2";
+package hw.trezor.messages.zcash;
+
+// Sugar for easier handling in Java
+option java_package = "com.satoshilabs.trezor.lib.protobuf";
+option java_outer_classname = "TrezorMessageZcash";
+
+/**
+ * Receiver typecodes for unified addresses.
+ * see: https://zips.z.cash/zip-0316#encoding-of-unified-addresses
+ */
+enum ZcashReceiverTypecode {
+  P2PKH = 0;
+  P2SH = 1;
+  SAPLING = 2;
+  ORCHARD = 3;
+}
```

Comment:
Why do we need this `enum` in a protobuf definition? As far as I am able to tell this is never communicated over USB, so it could be defined more locally in `unified_addresses.py`. Will it be needed in the future in the protobuf messages? If yes, then how exactly?

Reply:
The only place that I can think of where this could be used is for unified addresses in `GetAddress` if there would be a repeated field for the receivers. In that case I wonder if it wouldn't be better to recycle `InputScriptType` for this purpose in `GetAddress`.

Reply:
No. This is not exposed and it will never be exposed. `t_address_n` and `z_address_n` fields will be used for getting unified addresses.

I will move this enum into unified_addresses.py

Reply:
[5f33136](https://github.com/trezor/trezor-firmware/pull/2398/commits/5f33136b275176497f1e72d4651beb1aa3ce7d1d)